rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java

```diff
@@ -104,25 +108,51 @@ public int executeInNewJvm(String parameters, String pathToPom, String testParam
 			command.add("-o");
 		}
 		ProcessBuilder processBuilder = new ProcessBuilder();
-		processBuilder.inheritIO();
 		processBuilder.directory(new File(pathToPom));
-		processBuilder.redirectErrorStream(true);
 		logger.info("Executing maven in new jvm with command={}", command);
 		processBuilder.command(command);
 		Process process = processBuilder.start();
+		Thread outputCopyThread = createOutputCopyThread(process.getInputStream(), out);
+		outputCopyThread.run();
 		try {
 			while (!process.waitFor(100, TimeUnit.MILLISECONDS)) {
 				if (monitor.isCanceled()) {
+					outputCopyThread.interrupt();
 					process.destroy();
 				}
 			}
 			result = process.exitValue();
+			outputCopyThread.interrupt();
 		} catch (InterruptedException e) {
 			e.printStackTrace();
 		}
 		return result;
 	}
 
+	private Thread createOutputCopyThread(InputStream source, OutputStream out) {
+		return new Thread(new Runnable() {
+
+			@Override
+			public void run() {
+				BufferedReader reader = new BufferedReader(new InputStreamReader(source));
+				try {
+					String message = "";
+					while (message != null) {
+						out.write((message + "\n").getBytes());
+						out.flush();
+						logger.trace(message);
+						Thread.sleep(100);
+						message = reader.readLine();
+					}
+				} catch (IOException e) {
+					logger.error("Cant connect to process ouput stream", e);
+				} catch (InterruptedException e) {
+					logger.trace("Process log copy terminated.");
```

Comment:
Can we set a flag instead of working with `InterruptedException` here? I think this only triggers if the thread is currently sleeping. If it is just writing out stuff and the interrupt is triggered, what happens then?